src/components/App.js

```diff
@@ -67,10 +69,34 @@ class App extends Component {
 
   connectIfNeeded() {
     if (!this.props.pincodeRequired && this.props.shouldConnect) {
-      this.props.initConnection();
+      // this.props.initConnection();
+      this.onInitConnection();
     }
   }
 
+  onSetPincode = (pincode = "") => {
+    if (!pincode || pincode.endsWith("--")) {
+      this.onError('Invalid pincode!');
+      return;
+    }
+    // this.props.initConnection(pincode);
+    this.onInitConnection(pincode);
+  };
+
+
+  onInitConnection(pincode='') {
+    const urlHash = document.location.hash + pincode;
```

Comment:
Step 1. The first issue was that the very start of the observable auth flow had a dependency on the browser. This just injects the browser value into flow. When it's not the browser, ie a mobile device, it will come from a navigation screen parameter or something.